dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+#!/usr/bin/python
+""" Classes and functions for fitting tensors """
+from __future__ import division, print_function, absolute_import
+
+import warnings
+
+import numpy as np
+
+import scipy.optimize as opt
+
+from dipy.reconst.dti import (TensorFit, fractional_anisotropy,
+                              geodesic_anisotropy, mean_diffusivity,
+                              axial_diffusivity, radial_diffusivity, trace,
+                              color_fa, determinant, isotropic, deviatoric,
+                              norm, mode, linearity, planarity, sphericity,
+                              apparent_diffusion_coef, from_lower_triangular,
+                              lower_triangular, decompose_tensor)
+
+from dipy.sims.voxel import DKI_signal
+from dipy.utils.six.moves import range
+from dipy.data import get_sphere
+from ..core.gradients import gradient_table
+from ..core.geometry import vector_norm
+from ..core.sphere import Sphere
+from .vec_val_sum import vec_val_vect
+from ..core.onetime import auto_attr
+from .base import ReconstModel
+
+
+def rdpython(x,y,z):
+    r"""
+    WIP
+    """
+    d1mach=np.zeros(5)
+    d1mach[0]=1.1*10**(-308)
+    d1mach[1]=8.9e307
+    d1mach[2]=0.22204460*10**(-15)
+    d1mach[3]=0.4440*10**(-15)
+    d1mach[4]=np.log(2.0)
+    errtol = (d1mach[2]/3.0)**(1.0/6.0)
+    lolim  = 2.0/(d1mach[1])**(2.0/3.0)
+    tuplim = d1mach[0]**(1.0/3.0)
+    tuplim = (0.10*errtol)**(1.0/3.0)/tuplim
+    uplim  = tuplim**2.0
+    c1 = 3.0/14.0
+    c2 = 1.0/6.0
+    c3 = 9.0/22.0
+    c4 = 3.0/26.0
+
+    xn = x.copy()
+    yn = y.copy()
+    zn = z.copy()
+    sigma = 0.0
+    power4 = 1.0
+
+    mu = (xn+yn+3.0*zn)*0.20
+    xndev = (mu-xn)/mu
+    yndev = (mu-yn)/mu
+    zndev = (mu-zn)/mu
+    epslon = np.max([np.abs(xndev), np.abs(yndev), np.abs(zndev)])
+    while (epslon >= errtol):
+       xnroot = np.sqrt(xn)
+       ynroot = np.sqrt(yn)
+       znroot = np.sqrt(zn)
+       lamda = xnroot*(ynroot+znroot) + ynroot*znroot
+       sigma = sigma + power4/(znroot*(zn+lamda))
+       power4 = power4*0.250
+       xn = (xn+lamda)*0.250
+       yn = (yn+lamda)*0.250
+       zn = (zn+lamda)*0.250
+       mu = (xn+yn+3.0*zn)*0.20
+       xndev = (mu-xn)/mu
+       yndev = (mu-yn)/mu
+       zndev = (mu-zn)/mu
+       epslon = np.max([np.abs(xndev), np.abs(yndev), np.abs(zndev)])
+
+    ea = xndev*yndev
+    eb = zndev*zndev
+    ec = ea - eb
+    ed = ea - 6.0*eb
+    ef = ed + ec + ec
+    s1 = ed*(-c1+0.250*c3*ed-1.50*c4*zndev*ef)
+    s2 = zndev*(c2*ef+zndev*(-c3*ec+zndev*c4*ea))
+    drd = 3.0*sigma + power4*(1.0+s1+s2)/(mu*np.sqrt(mu))
+    return drd
+
+
+def rfpython(x,y,z):
```

Comment:
This function name is not clear. Is "python" really useful here? And what rf means?